components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtUtil.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.governance;
+
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataManagementService;
+
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.LocalClaim;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.governance.internal.IdentityMgtServiceDataHolder;
+import org.wso2.carbon.identity.governance.model.CustomPersistenceEnabledClaims;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utility class for Identity Claim Management related operations.
+ */
+public class IdentityMgtUtil {
+
+    private static final ClaimMetadataManagementService claimMetadataManagementService = IdentityMgtServiceDataHolder
+            .getInstance().getClaimMetadataManagementService();
+
+    /**
+     * Extract the user store persistency configured claims.
+     *
+     * @param claims           Array of claims to be filtered.
+     * @param userStoreManager UserStoreManager instance.
+     * @return Pair of Lists containing user store persisted and identity db persisted claims respectively.
+     */
+    public static CustomPersistenceEnabledClaims getCustomPersistenceEnabledClaims(
+            String[] claims, UserStoreManager userStoreManager) throws UserStoreException {
+
+        List<String> userStorePersistentClaims = new ArrayList<>();
+        List<String> identityStorePersistentClaims = new ArrayList<>();
+
+        // User store domain.
+        String userStoreDomain = userStoreManager.getRealmConfiguration().getUserStoreProperty(UserCoreConstants
+                .RealmConfig.PROPERTY_DOMAIN_NAME);
+
+        // Retrieve tenant domain.
+        String tenantDomain = getTenantDomain(userStoreManager);
```

Comment:
Let's get the tenant domain by this method.
PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain().